src/containers/profile/src/reducers/selectCaseReducer.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+import { Map } from 'immutable';
+
+import { ProfileReduxConstants } from '../../../../core/redux/constants';
+
+const { SELECTED_CASE } = ProfileReduxConstants;
+
+export default function reducer(state :Map, action :Object) {
+
+  const { value } = action;
+  const selectedCase :Map = value;
```

Comment:
Eh... this feels unnecessary. Flow isn't helping us here at all, so you can just do...

```
return state.set(SELECTED_CASE, action.value);
```

Reply:
i agree!